pkg/codegen/docs/gen.go

```diff
@@ -486,6 +486,40 @@ func (mod *modContext) withDocGenContext(dctx *docGenContext) *modContext {
 	return &newctx
 }
 
+func (dctx *docGenContext) getSupportedLanguages(isOverlay bool, overlaySupportedLanguages []string) []string {
```

Comment:
I feel like we could give this function a comment, which would obviate the need for the comments inside it too?